app/src/main/java/com/example/habitsmasher/DaysTracker.java

```diff
@@ -0,0 +1,261 @@
+     * storing, querying, modifying, and returning what days of the week a habit takes place. This class does not currently handle
+     * storing the days in a database, but is capable of taking in information stored in a database and converting to a DaysTracker object.
+     *
+     * @author: Cameron Matthew
+     * @Version: 1.0
+     * @see Habit
+     */
+
+    //instance variables to represent days of the week
+    private ArrayList<Boolean> _days;
+
+
+    /**
+     * Creates a DaysTracker object with all values set to false. This constructor is the default constructor.
+     */
+    public DaysTracker(){
+        //arraylist of booleans that represent the days of the week. autofilled to 7 bools.
+        _days = new ArrayList<Boolean>(Arrays.asList(new Boolean[7]));
+
+        //fill with all false
+        Collections.fill(_days, Boolean.FALSE);
+    }
+
+
+    /**
+     * Creates a DaysTracker with {@link String} input. The string contains all of the days that are to
+     * be set to true, and all other days false.
+     *<p>
+     * The string takes the format "day1 day2 ... day7", where each day is the first 2 letters of that day, separated by a space. The string can be in any order.
+     * EX) "MO FR WE TH"
+     *
+     * The string is assumed to be formatted correctly, but will not throw any exceptions if not correctly formatted. If any
+     * doubts are present when using this constructor, it may be wise to use the base constructor and set individual days yourself.
+     *
+     * @param days: The string of dates to use, formatted according to above guide.
+     */
+    public DaysTracker(String days){
+
+        _days = new ArrayList<Boolean>(Arrays.asList(new Boolean[7]));
+        Collections.fill(_days, Boolean.FALSE);
+
+        //take the string and split it up based on what days are present
+        //this will go under the assumption that the string will always be formatted correctly.
+        days = days.toUpperCase();
+        String[] splitDays = days.split(" ");
+
+        //for each item split from the string
+        for(int i = 0; i < splitDays.length; i++){
+            //get the value, and if any matches with the abbreviations set the corresponding day to true
+            String value = splitDays[i];
+            switch (value){
+                case "MO":
+                    _days.set(0, true);
+                    break;
+                case "TU":
+                    _days.set(1, true);
+                    break;
+                case "WE":
+                    _days.set(2, true);
+                    break;
+                case "TH":
+                    _days.set(3, true);
+                    break;
+                case "FR":
+                    _days.set(4, true);
+                    break;
+                case "SA":
+                    _days.set(5, true);
+                    break;
+                case "SU":
+                    _days.set(6, true);
+                    break;
```

Comment:
I think this chunk of code should be extracted out into it's own method and **unit tested**! This is exactly the type of logic we need to get under test.

Reply:
Has now been extracted and unit tested. The functionality was technically already tested with the string based constructor, but I added more tests apart from the constructor.